src/syft/core/node/domain/client.py

```diff
@@ -65,13 +65,67 @@ def __getitem__(self, key: Union[str, int]) -> RequestMessage:
     def __repr__(self) -> str:
         return repr(self.requests)
 
-    def add_handler(self, request_handler: Dict[str, Any]) -> None:
-        self.update_handler(request_handler, keep=True)
+    def add_handler(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+        element_quota: Optional[int] = None,
+    ) -> None:
+        handler_opts = self._validate_options(
+            action=action,
+            print_local=print_local,
+            log_local=log_local,
+            name=name,
+            timeout_secs=timeout_secs,
+            element_quota=element_quota,
+        )
 
-    def remove_handler(self, request_handler: Dict[str, Any]) -> None:
-        self.update_handler(request_handler, keep=False)
+        self._update_handler(handler_opts, keep=True)
 
-    def update_handler(self, request_handler: Dict[str, Any], keep: bool) -> None:
+    def remove_handler(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+    ) -> None:
+        handler_opts = self._validate_options(
+            action=action,
+            print_local=print_local,
+            log_local=log_local,
+            name=name,
+            timeout_secs=timeout_secs,
+        )
+
+        self._update_handler(handler_opts, keep=False)
+
+    def _validate_options(
+        self,
+        action: str,
+        print_local: bool = False,
+        log_local: bool = False,
+        name: Optional[str] = None,
+        timeout_secs: int = -1,
+        element_quota: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        handler_opts: Dict[str, Any] = {}
+        if action not in ["accept", "deny"]:
+            raise Exception("Action must be 'accept' or 'deny'")
+        handler_opts["action"] = action
+        handler_opts["print_local"] = bool(print_local)
+        handler_opts["log_local"] = bool(log_local)
+        if name is not None and name != "":
+            handler_opts["name"] = str(name)
+        handler_opts["timeout_secs"] = max(-1, int(timeout_secs))
+        if element_quota is not None:
+            handler_opts["element_quota"] = max(0, int(element_quota))
```

Comment:
Q: Why cast to int?

Reply:
I guess im paranoid that a non int sneaks through here and breaks the code preventing the quota from working. We should add the type checking decorator too but I don't trust python to deliver an int there unless we force it to.
Heres 1 horrible hack example:
```
>>> max(0, float("inf"))
inf
```

vs

```
>>> max(0, int(float("inf")))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: cannot convert float infinity to integer
```